File: IOS/DTOs/Objects/Customer/Catalog.swift
import Foundation

public enum Catalog {
    public enum Responses {}
}

public extension Catalog.Responses {
    
    struct Catalog: Codable, Sendable {
        public let services: [Service.Helpers.ServiceResponse]
        public let procedures: [Procedure.Helpers.AllProcedureResponse]
        public let complexes: [Complex.Helpers.ComplexResponse]
        
        public init(
            services: [Service.Helpers.ServiceResponse],
            procedures: [Procedure.Helpers.AllProcedureResponse],
            complexes: [Complex.Helpers.ComplexResponse]
        ) {
            self.services = services
            self.procedures = procedures
            self.complexes = complexes
        }
    }
}

File: android-legacy/shared/src/main/kotlin/shared/dto/objects/Catalog.kt
package shared.dto.objects

import kotlinx.serialization.Serializable
import shared.dto.objects.procedure.Complex
import shared.dto.objects.procedure.Procedure
import shared.dto.objects.procedure.Service
import shared.dto.protocols.Parametable
import shared.dto.protocols.Responsable
import java.util.*

object Catalog {
    data object Responses {
        @Serializable
        data class Retrieve(
            val services: List<Service.Helpers.ServiceResponse>,
            val procedures: List<Procedure.Helpers.AllProcedureResponse>,
            val complexes: List<Complex.Helpers.ComplexResponse>,
        ) : Responsable
    }
}
